HW2/P2/P2.py

```diff
@@ -74,9 +90,23 @@
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
```

Comment:
Good choice of range of locks and averaging the time to generate a clear trend plot. For the correlated and uncorrelated code, most are repetitive, maybe good to combine and make it more concise.